src/Models/Model.php

```diff
@@ -51,12 +55,45 @@ abstract class Model implements ModelInterface, Arrayable, JsonSerializable {
 	 *
 	 * @param array<string,mixed> $attributes Attributes.
 	 */
-	public function __construct( array $attributes = [] ) {
+	final public function __construct( array $attributes = [] ) {
```

Comment:
The last thing that i would ask is for developers to have a way to implement their own Model logic during the constructor. The final prevents them of that, so we could add a protected method for just that.

```php
$this->syncOriginal();
$this->duringBuild();
```

Reply:
Good idea! I added `afterConstruct` in 18c9152